qa/src/test/java/org/eclipse/kapua/service/datastore/steps/DataStoreServiceSteps.java

```diff
@@ -580,13 +703,88 @@ public void queryForTopicMetrics(String topic, String lstKey) throws KapuaExcept
         stepData.put(lstKey, tmpList);
     }
 
+    @When("^I query for the metrics from client \"(.*)\" and store them as \"(.*)\"$")
+    public void queryForClientMetrics(String clientId, String lstKey) throws KapuaException {
+
+        Account account = (Account) stepData.get("LastAccount");
+        MetricInfoQuery tmpQuery = DatastoreQueryFactory.createBaseMetricInfoQuery(account.getId(), 100);
+        AndPredicate andPredicate = new AndPredicateImpl();
+        TermPredicate tmpPred = new TermPredicateImpl(MetricInfoField.CLIENT_ID, clientId);
+        MetricInfoListResult tmpList = null;
+
+        andPredicate.getPredicates().add(tmpPred);
+        tmpQuery.setPredicate(andPredicate);
+        tmpList = metricInfoRegistryService.query(tmpQuery);
+        stepData.put(lstKey, tmpList);
+    }
+
     @Then("^There (?:is|are) exactly (\\d+) metric(?:|s) in the list \"(.*)\"$")
     public void checkNumberOfQueriedMetrics(int cnt, String lstKey) {
 
         MetricInfoListResult tmpResults = (MetricInfoListResult) stepData.get(lstKey);
         assertEquals(cnt, tmpResults.getSize());
     }
 
+    @Then("^Client \"(.+)\" first published a metric in the list \"(.+)\" on \"(.+)\"$")
+    public void checkFirstPublishDateOfClientMetric(String clientId, String lstKey, String date) throws Exception {
+
+        MetricInfoListResult metList = (MetricInfoListResult) stepData.get(lstKey);
+        Date tmpCaptured = parseFeatureDateTimeString(date);
+
+        for (MetricInfo tmpMet : metList.getItems()) {
+            if (tmpMet.getClientId().equals(clientId)) {
+                assertEquals(tmpMet.getFirstMessageOn(), tmpCaptured);
+                return;
+            }
+        }
+        fail(String.format("No metric matches the client id [%s]", clientId));
+    }
+
+    @Then("^Client \"(.+)\" last published a metric in the list \"(.+)\" on \"(.+)\"$")
+    public void checkLastPublishDateOfClientMetric(String clientId, String lstKey, String date) throws Exception {
+
+        MetricInfoListResult metList = (MetricInfoListResult) stepData.get(lstKey);
+        Date tmpCaptured = parseFeatureDateTimeString(date);
```

Comment:
I was looking at that code and may I missed this check in a previous pr review.
The date parsing should be alligned to the Kapua datastore format so can you change this method implementation to use:
https://github.com/eclipse/kapua/blob/develop/commons/src/main/java/org/eclipse/kapua/commons/util/KapuaDateUtils.java#L57
With this change the incoming dates should be changed (according to the pattern)
May be better if we address this fix in a new pr, what do you think?